tests/testthat/test-slice-assign.R

```diff
@@ -280,10 +290,22 @@ test_that("slice-assign falls back to `[<-` when proxy is not implemented", {
   )
 
   obj <- foobar(c("foo", "bar", "baz"))
-  vec_slice(obj, 1:2) <- NA
+  vec_slice(obj, 1:2) <- TRUE
   expect_identical(obj, c("dispatched", "dispatched", "baz"))
 })
 
+test_that("vec_assign() cannot assign unspecified values into foreign vector types", {
+  obj <- foobar(c("foo", "bar", "baz"))
+
+  # These are correctly incompatible cast errors, not type errors.
```

Comment:
This should work if we hard-code casts of unspecified vectors to vec-init right? (provided `[` or proxy/restore are correctly implemented). I think we should go in that direction.

Reply:
What should we do about integer64? `vec_init()` won't work because `[.integer64` doesn't work with `NA_integer_`.

I think I agree with you that I think this should be hardcoded to `vec_init(to, vec_size(x))`, but that would break the integer64 case. Normally I'd say that that would be the fault of the package author, and not us, but otherwise it would probably break legacy code in dplyr/tidyr that uses integer64

Reply:
I've been thinking we should just special-case the correct behaviour for integer64.

Reply:
Fixed now that #813 is supported